packages/orchestration/src/exos/stakingAccountKit.js

```diff
@@ -39,16 +43,44 @@ const { Fail } = assert;
 const HolderI = M.interface('holder', {
   getPublicTopics: M.call().returns(TopicsRecordShape),
   makeDelegateInvitation: M.call(M.string(), AmountShape).returns(M.promise()),
+  makeWithdrawRewardInvitation: M.call(M.string()).returns(M.promise()),
   makeCloseAccountInvitation: M.call().returns(M.promise()),
   makeTransferAccountInvitation: M.call().returns(M.promise()),
-  delegate: M.callWhen(M.string(), AmountShape).returns(M.string()),
+  delegate: M.callWhen(M.string(), AmountShape).returns(M.record()),
 });
 
 /** @type {{ [name: string]: [description: string, valueShape: Pattern] }} */
 const PUBLIC_TOPICS = {
   account: ['Staking Account holder status', M.any()],
 };
 
+// UNTIL https://github.com/cosmology-tech/telescope/issues/605
+/**
+ * @param {Any} x
+ * @returns {AnyJson}
+ */
+const toAnyJSON = x => /** @type {AnyJson} */ (Any.toJSON(x));
+
+/**
+ * @template T
+ * @param {string} ackStr
+ * @param {(p: {typeUrl: string, value: Uint8Array}) => T} fromProtoMsg
+ */
+export const tryDecodeResponse = (ackStr, fromProtoMsg) => {
+  try {
+    const any = Any.decode(decodeBase64(ackStr));
+    const protoMsg = Any.decode(any.value);
+
+    const msg = fromProtoMsg(protoMsg);
+    return msg;
+  } catch (cause) {
+    throw assert.error(`bad response: ${ackStr}`, undefined, { cause });
+  }
+};
```

Comment:
nb: would be nice to see this pulled out of this file as it seems reusable. This seems like it would be consumer facing for folks implementing custom messages.